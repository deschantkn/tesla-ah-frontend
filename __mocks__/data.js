/* eslint-disable import/prefer-default-export */
export const article = [
  {
    id: 16,
    slug: 'human-errors-book-review-rising-above-our-weaknesses-xjzipva6x',
    title: 'Human Errors book review: Rising above our weaknesses',
    readtime: '2 min',
    description:
      'WHETHER we believe in intelligent design or put our trust in evolution, there is one thing we all more or less agree on: the human being is a pretty nifty creation. We have colonised every part of the planet, we are at the top of the food chain and our brains are magnificent.',
    body:
      '{"article":{"body":{"blocks":[{"key":"4d32s","text":"Human Errors book review: Rising above our weaknesses","type":"header-two","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"a7lod","text":"WHETHER we believe in intelligent design or put our trust in evolution, there is one thing we all more or less agree on: the human being is a pretty nifty creation. We have colonised every part of the planet, we are at the top of the food chain and our brains are magnificent.","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"unoc","text":"","type":"image","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{"aspect_ratio":{"width":386,"height":131,"ratio":33.93782383419689},"width":386,"caption":"type a caption (optional)","height":131,"forceUpload":false,"url":"https://res.cloudinary.com/newpoint/image/upload/v1566071168/lksj3nq8aygqcblnnk2v.jpg","loading_progress":0,"selected":false,"loading":true,"file":null,"direction":"center"}},{"key":"84pu4","text":"n fact, whether it is by divine writ or Darwinian randomness, humans are the fittest of the fit and the best of all possible organisms.","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"197q1","text":"In Human Errors, Nathan Lents explains in a chatty and humorous style all the defects that the human genome has saddled us with, from the inefficient shape of our eyeballs to the excessive number of bones in our ankles.","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"3lg1c","text":"","type":"image","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{"aspect_ratio":{"width":641,"height":209,"ratio":32.605304212168484},"width":641,"caption":"type a caption (optional)","height":209,"forceUpload":false,"url":"https://res.cloudinary.com/newpoint/image/upload/v1566071231/srsdeywc83dhbzwl71io.jpg","loading_progress":0,"selected":false,"loading":true,"file":null,"direction":"center"}},{"key":"ee2ie","text":"One of the main problems with evolution and the slow progress it makes through random mutation is that it saddles us with body parts and functions that may once have served a useful purpose but are no longer fit for purpose.","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"2ek0i","text":"One of the main problems with evolution and the slow progress it makes through random mutation is that it saddles us with body parts and functions that may once have served a useful purpose but are no longer fit for purpose.","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"4a2t2","text":"random mutation is that it saddles us with body parts and functions that may once have served a useful purpose but are no longer fit for purpose.","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}]},"title":"Human Errors book review: Rising above our weaknesses","description":"Human Errors book review: Rising above our weaknesses","tagList":["Human","Errors"]}}',
    blocked: false,
    tagList: ['Human', 'Errors', 'Weakness'],
    gallery: null,
    authorId: 21,
    views: 225,
    createdAt: '2019-08-17T19:48:06.166Z',
    updatedAt: '2019-08-20T10:40:57.911Z',
  },
  {
    id: 4,
    slug: 'how-to-create-an-api-in-node-js-and-express-4cjjz6uyzwr',
    title: 'How to create an API in node js and express',
    readtime: '2 min',
    description:
      'I belong to what you would classify as the upper class of the Pakistani community. Well educated, wealthy, top military brass. A family of physicians, engineers, and senior management of leading corporations. My parents had lived two decades in the U.K. and Europe, where my siblings and I were raised. My husband’s family is a family entirely of physicians.',
    body:
      '{"article":{"body":{"blocks":[{"key":"4d32s","text":"How to create an API in node js and express","type":"header-two","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"5pu4m","text":"I belong to what you would classify as the upper class of the Pakistani community. Well educated, wealthy, top military brass. A family of physicians, engineers, and senior management of leading corporations. My parents had lived two decades in the U.K. and Europe, where my siblings and I were raised. My husband’s family is a family entirely of physicians.","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"3c6hk","text":"image","type":"image","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{"aspect_ratio":{"width":750,"height":422,"ratio":56.266666666666666},"width":750,"caption":"type a caption (optional)","height":422,"forceUpload":false,"url":"https://res.cloudinary.com/tesla-ah-media/image/upload/v1565521197/aohpzzejvcxwol3rpjma.jpg","loading_progress":0,"selected":false,"loading":true,"file":null,"direction":"center"}},{"key":"dgv0t","text":"I belong to what you would classify as the upper class of the Pakistani community. Well educated, wealthy, top military brass. A family of physicians, engineers, and senior management of leading corporations. My parents had lived two decades in the U.K. and Europe, where my siblings and I were raised. My husband’s family is a family entirely of physicians.","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"c0ae4","text":"node","type":"image","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{"aspect_ratio":{"width":1000,"height":380.859375,"ratio":38.0859375},"width":1024,"caption":"type a caption (optional)","height":390,"forceUpload":false,"url":"https://res.cloudinary.com/tesla-ah-media/image/upload/v1565521227/wdshglthatgzp5sdt2a0.jpg","loading_progress":0,"selected":false,"loading":true,"file":null,"direction":"center"}},{"key":"bqum4","text":"","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"8i2re","text":"I belong to what you would classify as the upper class of the Pakistani community. Well educated, wealthy, top military brass. A family of physicians, engineers, and senior management of leading corporations. My parents had lived two decades in the U.K. and Europe, where my siblings and I were raised. My husband’s family is a family entirely of physicians.","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"1co84","text":"I belong to what you would classify as the upper class of the Pakistani community. Well educated, wealthy, top military brass. A family of physicians, engineers, and senior management of leading corporations. My parents had lived two decades in the U.K. and Europe, where my siblings and I were raised. My husband’s family is a family entirely of physicians.","type":"blockquote","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}]},"title":"How to create an API in node js and express","description":"How to create an API in node js and express","tagList":[]}}',
    blocked: false,
    tagList: ['Andela', 'Technology', 'Math'],
    gallery: [],
    authorId: 5,
    views: 0,
    createdAt: '2019-08-11T11:01:57.963Z',
    updatedAt: '2019-08-11T11:01:57.963Z',
  },
  {
    id: 5,
    slug: 'how-to-create-an-api-in-node-js-and-express-5n1jz703vof',
    title: 'How to create an API in node js and express',
    readtime: 'Less than a minute',
    description:
      'I belong to what you would classify as the upper class of the Pakistani community. Well educated, wealthy, top military brass. A family of physicians, engineers, and senior management of leading corporations. My parents had lived two decades in the U.K. and Europe, where my siblings and I were raised. My husband’s family is a family entirely of physicians.',
    body:
      '{"article":{"body":{"blocks":[{"key":"4d32s","text":"How to create an API in node js and express","type":"header-two","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"ecluj","text":"I belong to what you would classify as the upper class of the Pakistani community. Well educated, wealthy, top military brass. A family of physicians, engineers, and senior management of leading corporations. My parents had lived two decades in the U.K. and Europe, where my siblings and I were raised. My husband’s family is a family entirely of physicians.","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"emitn","text":"I belong to what you would classify as the upper class of the Pakistani community. Well educated, wealthy, top military brass. A family of physicians, engineers, and senior management of leading corporations. My parents had lived two decades in the U.K. and Europe, where my siblings and I were raised. My husband’s family is a family entirely of physicians.","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"3oi1u","text":"","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}]},"title":"How to create an API in node js and express","description":"How to create an API in node js and express","tagList":[]}}',
    blocked: false,
    tagList: ['Andela', 'Technology'],
    gallery: [],
    authorId: 5,
    views: 0,
    createdAt: '2019-08-11T13:25:43.840Z',
    updatedAt: '2019-08-11T13:25:43.840Z',
  },
  {
    id: 6,
    slug:
      'googles-newest-gadgets-are-bringing-science-fiction-to-real-life-65gjz712jiw',
    title: 'Google’s newest gadgets are bringing science fiction to real life.',
    readtime: '2 min',
    description:
      'The tech giant’s lab for advanced projects unveiled some of its hottest new innovations last Friday, which included finger control technology and a pair of smart jeans. As with all things Google, they proved extremely impressive.',
    body:
      '{"article":{"body":{"blocks":[{"key":"4d32s","text":"Google’s newest gadgets are bringing science fiction to real life.","type":"header-two","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"841cl","text":"The tech giant’s lab for advanced projects unveiled some of its hottest new innovations last Friday, which included finger control technology and a pair of smart jeans. As with all things Google, they proved extremely impressive.","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"3e82h","text":"We’ll start off with the finger control technology. Dubbed Project Soli, the gadget takes the form of a radar that can detect precise hand movements and translate them into commands. In other words, it’s just like touch screen technology, only it doesn’t require the screen. During a showcase, Google presenters were able to kick a virtual soccer ball by flicking their fingers, or shift the hours on a clock by turning an imaginary dial.","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":59,"length":12,"style":"ITALIC"}],"entityRanges":[],"data":{}},{"key":"1ffd7","text":"image","type":"image","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{"aspect_ratio":{"width":320,"height":160,"ratio":50},"width":320,"caption":"type a caption (optional)","height":160,"forceUpload":false,"url":"https://res.cloudinary.com/tesla-ah-media/image/upload/v1565531346/k5jqfj07rverjw5bqqk4.gif","loading_progress":0,"selected":false,"loading":true,"file":null,"direction":"center"}},{"key":"cip44","text":"We’ll start off with the finger control technology. Dubbed Project Soli, the gadget takes the form of a radar that can detect precise hand movements and translate them into commands. In other words, it’s just like touch screen technology, only it doesn’t require the screen. During a showcase, Google presenters were able to kick a virtual soccer ball by flicking their fingers, or shift the hours on a clock by turning an imaginary dial.","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":59,"length":12,"style":"ITALIC"}],"entityRanges":[],"data":{}},{"key":"9iq9j","text":"image","type":"image","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{"aspect_ratio":{"width":500,"height":215,"ratio":43},"width":500,"caption":"type a caption (optional)","height":215,"forceUpload":false,"url":"https://res.cloudinary.com/tesla-ah-media/image/upload/v1565531386/utkia6gp8p0qkqxxnnti.gif","loading_progress":0,"selected":false,"loading":true,"file":null,"direction":"center"}},{"key":"90i4i","text":"We’ll start off with the finger control technology. Dubbed Project Soli, the gadget takes the form of a radar that can detect precise hand movements and translate them into commands. In other words, it’s just like touch screen technology, only it doesn’t require the screen. During a showcase, Google presenters were able to kick a virtual soccer ball by flicking their fingers, or shift the hours on a clock by turning an imaginary dial.","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":59,"length":12,"style":"ITALIC"}],"entityRanges":[],"data":{}},{"key":"cd71j","text":"https://youtu.be/0QNiZfSsPc0","type":"video","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{"provisory_text":"https://youtu.be/0QNiZfSsPc0","endpoint":"//noembed.com/embed?url=","type":"video","embed_data":{"provider_url":"https://www.youtube.com/","width":480,"height":270,"html":"\\n<iframe width=\\" 480\\" height=\\"270\\" src=\\"https://www.youtube.com/embed/0QNiZfSsPc0?feature=oembed\\" frameborder=\\"0\\" allowfullscreen=\\"allowfullscreen\\"></iframe>\\n","url":"https://youtu.be/0QNiZfSsPc0","thumbnail_width":480,"version":"1.0","title":"Welcome to Project Soli","provider_name":"YouTube","type":"video","thumbnail_height":360,"author_url":"https://www.youtube.com/user/GoogATAP","thumbnail_url":"https://i.ytimg.com/vi/0QNiZfSsPc0/hqdefault.jpg","author_name":"Google ATAP"}}},{"key":"1ff59","text":"As for Google’s smart jeans, they’re about to blow your socks off. The company partnered with Levis and found a way to engineer conductive cloth that it says is “indistinguishable” from regular fabrics, and can be controlled by a chip the size of a button. That means Google can turn your pants or other clothing items into a touchpad","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"7k4le","text":"Strangely, Google doesn’t really have many ideas for this new technology. The company is calling it a “blank canvas” for designers and developers, so it looks like the innovation will be up to third parties. The search engine giant is simply trying to perfect the art of marrying technology with clothing in a seamless way. You can check out the new jeans, codenamed Project Jacquard below:","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":367,"length":16,"style":"ITALIC"}],"entityRanges":[],"data":{}},{"key":"f9e45","text":"","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"t5f6","text":"https://youtu.be/qObSFfdfe7I","type":"video","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{"provisory_text":"https://youtu.be/qObSFfdfe7I","endpoint":"//noembed.com/embed?url=","type":"video","embed_data":{"provider_url":"https://www.youtube.com/","width":480,"height":270,"html":"\\n<iframe width=\\" 480\\" height=\\"270\\" src=\\"https://www.youtube.com/embed/qObSFfdfe7I?feature=oembed\\" frameborder=\\"0\\" allowfullscreen=\\"allowfullscreen\\"></iframe>\\n","url":"https://youtu.be/qObSFfdfe7I","thumbnail_width":480,"version":"1.0","title":"Welcome to Project Jacquard","provider_name":"YouTube","type":"video","thumbnail_height":360,"author_url":"https://www.youtube.com/user/GoogATAP","thumbnail_url":"https://i.ytimg.com/vi/qObSFfdfe7I/hqdefault.jpg","author_name":"Google ATAP"}}}]},"title":"Google’s newest gadgets are bringing science fiction to real life.","description":"Google’s newest gadgets are bringing science fiction to real life.","tagList":[]}}',
    blocked: false,
    tagList: ['Andela ', 'Technology', 'Google'],
    gallery: [],
    authorId: 5,
    views: 0,
    createdAt: '2019-08-11T13:52:41.050Z',
    updatedAt: '2019-08-11T13:52:41.050Z',
  },
];
