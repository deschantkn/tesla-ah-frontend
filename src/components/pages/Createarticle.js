/* eslint-disable react/no-unused-state */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import Editor from 'prestein-dante';
import PropTypes from 'prop-types';
import { toast } from 'react-toastify';
import { Redirect } from 'react-router-dom';
import EditorState from '../widgets/Editorstate';
import {
  uploadImage,
  createArticle,
  updateArticle,
} from '../../redux/actions/article.actions';
import { getFromEditor } from '../../utils/getArticleItemData';

export class Createarticle extends Component {
  state = {
    images: [],
    redirect: false,
    slug: '',
    noImageUploaded: true,
    noArticleCreated: true,
  };

  componentWillReceiveProps(newProps) {
    const { NewUploadedImage, NewArticle } = newProps;
    const currentUploadImagePosition = Editor.getCurrentUploadImagePos();
    if (NewUploadedImage.secure_url) {
      const { images } = this.state;
      images.push({
        position: currentUploadImagePosition,
        url: NewUploadedImage.secure_url,
      });
      this.setState({ images });
    } else {
      this.setState({ noImageUploaded: true });
    }
    if (NewArticle.article) {
      const { slug } = NewArticle.article;
      toast.success('Article published successful');
      this.setState({ redirect: true, slug });
    } else {
      this.setState({ noArticleCreated: true });
    }
  }

  render() {
    const { slug } = this.state;
    if (this.state.redirect) {
      return <Redirect to={`/articles/${slug}`} />;
    }
    return (
      <div className="container create-article-container">
        <Editor
          tags={[]}
          slug=""
          images={this.state.images}
          edtrState={EditorState}
          isUpdating={false}
          uploadImage={this.props.uploadImage}
          createArticle={this.props.createArticle}
          updateArticle={this.props.updateArticle}
          getFromEditor={getFromEditor}
        />
      </div>
    );
  }
}

Editor.propTypes = {
  uploadImage: PropTypes.func.isRequired,
  createArticle: PropTypes.func.isRequired,
  edtrState: PropTypes.instanceOf(Object).isRequired,
  updateArticle: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  NewUploadedImage: state.article.uploadedImage,
  NewArticle: state.article.currentArticle,
});

export default connect(
  mapStateToProps,
  { updateArticle, uploadImage, createArticle },
)(Createarticle);
