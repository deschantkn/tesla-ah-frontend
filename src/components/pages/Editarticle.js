/* eslint-disable no-unused-vars */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import Editor from 'prestein-dante';
import { toast } from 'react-toastify';
import { Redirect } from 'react-router-dom';
import Preloader from '../widgets/Preloader';
import {
  getArticle,
  uploadImage,
  createArticle,
  updateArticle,
  resetProps,
} from '../../redux/actions/article.actions';
import { getFromEditor } from '../../utils/getArticleItemData';

export class Editarticle extends Component {
  state = {
    Article: {},
    slug: '',
    redirect: false,
    images: [],
  };

  componentWillMount() {
    const { slug } = this.props.match.params;
    this.setState({ slug });
    this.props.getArticle(slug);
  }

  componentWillReceiveProps(newProps) {
    const { NewUploadedImage, UpdateRes } = newProps;
    const currentUploadImagePosition = Editor.getCurrentUploadImagePos();
    if (newProps.Article) {
      this.setState({ Article: newProps.Article });
    }
    if (NewUploadedImage.secure_url) {
      toast.success('Image uploaded successfully');
      const { images } = this.state;
      images.push({
        position: currentUploadImagePosition,
        url: NewUploadedImage.secure_url,
      });
      this.setState({ images });
    }
    if (UpdateRes.message) {
      toast.success(newProps.UpdateRes.message);
      this.setState({ redirect: true });
      this.props.resetProps();
    }
  }

  render() {
    const { Article, slug } = this.state;
    if (this.state.redirect) {
      return <Redirect to="/articles" />;
    }
    if (Article.status) {
      const content = JSON.parse(Article.article.body);
      const { blocks } = content.article.body;
      const tags = Article.article.tagList;
      return (
        <div className="container create-article-container">
          <Editor
            isUpdating
            tags={tags}
            slug={slug}
            images={this.state.images}
            edtrState={{ blocks, entityMap: {} }}
            uploadImage={this.props.uploadImage}
            createArticle={this.props.createArticle}
            updateArticle={this.props.updateArticle}
            getFromEditor={getFromEditor}
          />
        </div>
      );
    }
    return (
      <div>
        <Preloader />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  Article: state.article.article,
  UpdateRes: state.article.updatedArticle,
  NewUploadedImage: state.article.uploadedImage,
});
export default connect(
  mapStateToProps,
  {
    getArticle, updateArticle, uploadImage, createArticle, resetProps,
  },
)(Editarticle);
