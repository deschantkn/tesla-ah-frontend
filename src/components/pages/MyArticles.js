/* eslint-disable max-len */
/* eslint-disable no-unused-vars */
/* eslint-disable react/button-has-type */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/img-redundant-alt */
/* eslint-disable jsx-a11y/label-has-for */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import moment from 'moment';
import { toast } from 'react-toastify';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import Pagination from 'react-pagination-library';
import { fontFace } from 'polished';
import {
  getMyArticles,
  deleteArticle,
} from '../../redux/actions/article.actions';
import { setLoading, setLoaded } from '../../redux/actions/ui.actions';
import Preloader from '../widgets/Preloader';
import Image from '../items/Imageitem';
import { getItemDataFromDatabase } from '../../utils/getArticleItemData';

export class MyArticles extends Component {
  state = {
    currentPage: 1,
  };

  componentDidMount() {
    this.props.getMyArticles(this.state.currentPage);
    toast.success('Fetching Articles succeeded...');
  }

  changeCurrentPage = (numPage) => {
    this.setState({ currentPage: numPage });
    this.props.getMyArticles(this.state.currentPage);
  };

  render() {
    const foundArticles = this.props.articles.data;
    // console.log('foundArticles in redux', foundArticles);
    if (foundArticles === undefined) {
      return (
        <div>
          <div className="homecontainer" data-test="homeComponent">
            <Preloader />
          </div>
        </div>
      );
    }
    if (foundArticles) {
      const articles = foundArticles.foundArticles;
      return (
        <div className="container mt-5 mb-5 center my-articles-container">
          <br />
          <div className="row ml-2">
            <div className="col-lg-12">
              <div
                htmlFor="story_title"
                className="h3 font-weight-bold text-dark"
              >
                { foundArticles.count !== 0 ? 'My Stories' : (
                  <div className="row">
                    <div className="col-lg-3" />
                    <div className="col-lg-6 text-center text-third">
                      <h2>You don&apos;t have any article so far!</h2>
                    </div>
                    <div className="col-lg-3" />
                  </div>
                )}
              </div>
            </div>
          </div>
          {articles.map((article) => {
            const {
              image,
              body,
              article: {
                title, slug, createdAt, readtime,
              },
            } = getItemDataFromDatabase(article);
            return (
              <Link
                to={`/articles/${article.slug}`}
                style={{ textDecoration: 'none' }}
              >
                <div className="row mt-5 mb-3 ml-3 " key={article.slug}>
                  <div className="col-lg-7" style={{ cursor: 'pointer' }}>
                    <div className="row">
                      <label
                        htmlFor="story_title"
                        className="h5 text-dark font-weight-bold"
                      >
                        {title}
                      </label>
                    </div>
                    <div className="row">
                      <label
                        htmlFor="story_description"
                        className="text-secondary font-weight-light"
                      >
                        {body}
                      </label>
                    </div>
                    <div className="row">
                      <div className="col-lg-0">
                        <label
                          htmlFor="story_date"
                          className="h6 text-secondary font-weight-bold"
                        >
                          {moment(createdAt).format('DD MMMM YYYY')}
                        </label>
                      </div>
                      <div className="col-lg-6">
                        <label
                          htmlFor="story_readtime"
                          className="h6 text-secondary font-weight-normal story_readtime"
                        >
                          {readtime}
                        </label>
                      </div>
                      <div className="col-lg-8" />
                    </div>
                  </div>
                  <Image title={title} image={image} />
                </div>
              </Link>
            );
          })}
          <div className="text-center container pagination-container">
            <Pagination
              currentPage={this.state.currentPage}
              totalPages={Math.ceil(foundArticles.count / 5)}
              changeCurrentPage={this.changeCurrentPage}
            />
          </div>
        </div>
      );
    }
    return (
      <div>
        <div className="homecontainer" data-test="homeComponent">
          <Preloader />
        </div>
      </div>
    );
  }
}
export const mapStateToProps = state => ({
  articles: state.article.myarticles,
  ui: state.ui,
  login: state.login,
});
export default connect(
  mapStateToProps,
  {
    getMyArticles,
    setLoaded,
    setLoading,
    deleteArticle,
  },
)(MyArticles);
