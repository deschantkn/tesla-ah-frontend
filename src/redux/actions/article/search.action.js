import axios from 'axios';
// import { toast } from 'react-toastify';
import {
  SEARCH_PENDING, SEARCH_DONE, SEARCH_FAILED, SEARCH_CLEAR,
} from '../types/article.type';
import { BACKEND_URL } from '../../../utils/constants';

export default (queryText, type) => async (dispatch) => {
  dispatch({
    type: SEARCH_CLEAR,
  });
  dispatch({
    type: SEARCH_PENDING,
  });
  try {
    const { data: response } = await axios.get(`${BACKEND_URL}/api/articles/?${type}=${queryText}`);
    dispatch({
      type: SEARCH_DONE,
      payload: response.data,
    });
  } catch ({ response: { data } }) {
    dispatch({
      type: SEARCH_FAILED,
      payload: data.error,
    });
  }
};
