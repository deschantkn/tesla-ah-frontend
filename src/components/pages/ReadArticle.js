/* eslint-disable react/no-unused-state */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import DisplayContent from 'Dante2';
import Rater from 'react-rater';
import Moment from 'react-moment';
import { toast } from 'react-toastify';
import jwt from 'jwt-decode';
import { Link, Redirect } from 'react-router-dom';
import { getArticle, deleteArticle, resetProps } from '../../redux/actions/article.actions';
import { getUserProfile } from '../../redux/actions/author/authoruser.action';
import { DEFAULT_AVATA } from '../../utils/constants';
import Preloader from '../widgets/Preloader';


class ReadArticle extends Component {
  state = {
    Article: {},
    Author: {},
    user: {},
    slug: '',
    redirect: false,
  };

  componentWillMount() {
    const { slug } = this.props.match.params;
    let user = {};
    this.props.getArticle(slug);
    if (sessionStorage.getItem('token')) {
      user = jwt(sessionStorage.getItem('token'));
    }
    this.setState({ user, slug });
  }

  componentWillReceiveProps(newProps) {
    if (newProps.Article) {
      this.setState({ Article: newProps.Article });
      this.props.getUserProfile(newProps.Article.article.author.username);
    }
    if (newProps.Author) {
      this.setState({ Author: newProps.Author });
    }
    if (newProps.Delete.message) {
      this.setState({ redirect: true });
      toast.success('Articles has been deleted successfully...');
    }
  }

  render() {
    const {
      Article,
      Author,
      user: { username },
    } = this.state;
    let contentBlocks = [];
    if (this.state.redirect) {
      return <Redirect to="/articles" />;
    }
    if (Article && Author.profile) {
      const content = JSON.parse(Article.article.body);
      const { blocks } = content.article.body;
      contentBlocks = blocks.splice(1, blocks.length);
      const { avatar, firstName, lastName } = Author.profile;
      return (
        <div className="container view-article-content mt-5">
          <section className="editor-main-section">
            <main className="editor-main row mt-6 mb-5">
              <div className="col-lg-9 left-nav">
                <div className="title-content">
                  <p>
                    <strong>{blocks[0].text}</strong>
                  </p>
                </div>
                <div className="row profile-content ml-1">
                  <div className="">
                    <img className="" src={avatar || DEFAULT_AVATA} alt="" />
                  </div>
                  <div className="ml-3">
                    <div>
                      <strong>{`${firstName}  ${lastName}`}</strong>
                    </div>
                    <div>
                      <Moment format="D MMM YYYY">
                        {Article.article.createdAt}
                      </Moment>
                    </div>
                    <Rater total={5} rating={2} />
                  </div>
                </div>
                <div className="mt-3">
                  <DisplayContent
                    content={{ blocks: contentBlocks, entityMap: {} }}
                    read_only
                  />
                </div>
              </div>
              <div className="col-lg-1 rigth-nav text-center">
                <div className="social-buttons">
                  <div className="flauting-buttons mt-3 facebook">
                    <i className="fab fa-facebook-f" />
                  </div>
                  <div className="flauting-buttons mt-3 twitter">
                    <i className="fab fa-twitter" />
                  </div>
                  <div className="flauting-buttons mt-3 email">
                    <i className="fas fa-envelope-open" />
                  </div>
                  <div
                    className={`${
                      username === this.state.Article.article.author.username
                        ? 'hide-button'
                        : ''
                    } flauting-buttons mt-3 bookmark`}
                  >
                    <i className="far fa-bookmark" />
                  </div>
                  <div
                    className={`${
                      username === this.state.Article.article.author.username
                        ? 'hide-button'
                        : ''
                    } flauting-buttons mt-3`}
                  >
                    <i className="fas fa-thumbs-up" />
                  </div>
                  <div
                    className={`${
                      username === this.state.Article.article.author.username
                        ? 'hide-button'
                        : ''
                    } flauting-buttons mt-3 dislike`}
                  >
                    <i className="fas fa-thumbs-down" />
                  </div>
                  <Link to={`/article/edit/${this.state.slug}`}>
                    <div
                      className={`${
                        username !== this.state.Article.article.author.username
                          ? 'hide-button'
                          : ''
                      } flauting-buttons mt-3 edit`}
                    >
                      <i className="fas fa-edit" />
                    </div>
                  </Link>
                  <div
                    className={`${
                      username !== this.state.Article.article.author.username
                        ? 'hide-button'
                        : ''
                    } flauting-buttons mt-3 delete`}
                    data-toggle="modal"
                    data-target="#myModal"
                  >
                    <i className="fas fa-trash-alt" />
                  </div>
                </div>
              </div>
            </main>
            <div id="myModal" className="modal fade" role="dialog">
              <div className="modal-dialog">

                <div className="modal-content">
                  <div className="modal-header model-header-div">
                    <button type="button" className="close text-white" data-dismiss="modal">&times;</button>
                  </div>
                  <div className="modal-body modal-body-div">
                    <p> Are you sure you want to delete this article ?</p>
                  </div>
                  <div className="modal-footer modal-footer-div">
                    <button type="button" className="btn btn-default text-white" data-dismiss="modal">No</button>
                    <button type="button" className="btn btn-default text-white" data-dismiss="modal" onClick={() => this.props.deleteArticle(this.state.slug)}>Yes</button>
                  </div>
                </div>

              </div>
            </div>
          </section>
        </div>
      );
    }
    return (
      <div>
        <Preloader />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  Article: state.article.article,
  Author: state.author.authorprofile,
  Delete: state.article.deletedArticle,
});

export default connect(
  mapStateToProps,
  {
    getArticle, getUserProfile, deleteArticle, resetProps,
  },
)(ReadArticle);
