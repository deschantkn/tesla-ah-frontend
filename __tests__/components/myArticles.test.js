/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import moxios from 'moxios';
import configureStore from 'redux-mock-store';
import { MyArticles, mapStateToProps } from '../../src/components/pages/MyArticles';
import { getMyArticles } from '../../src/redux/actions/article.actions';
import { article } from '../../__mocks__/data';

const middleware = [thunk];
const mockStore = configureStore(middleware);
const store = mockStore({});

const props1 = {
  articles: {
    data: {
      foundArticles: [article[0], article[1]],
      count: 5,
    },
  },
  history: [],
  getMyArticles: jest.fn(),
  deleteArticle: jest.fn(),
};
const props2 = {
  articles: {},
  history: [],
  getMyArticles: jest.fn(),
  deleteArticle: jest.fn(),
};

describe('Fetch All Articles tests...', () => {
  const allArticles = shallow(<MyArticles {...props1} />);
  const allArticles1 = shallow(<MyArticles {...props2} />);
  it('it should render the AllArticles Component', () => {
    mapStateToProps({ article: { MyArticles: [] }, ui: {}, login: {} });
    expect(allArticles).toMatchSnapshot();
  });
  it('it should render the AllArticles Component', () => {
    allArticles1.instance().changeCurrentPage(1);
    expect(allArticles1).toMatchSnapshot();
  });
});

describe('Should get MyArticles', () => {
  beforeEach(() => {
    moxios.install();
  });
  afterEach(() => {
    moxios.uninstall();
    store.clearActions();
  });
  test('should get My Articles', async () => {
    await moxios.wait(() => {
      const request = moxios.requests.mostRecent();
      request.respondWith({
        status: 200,
        response: {
          article: { articles: [], TotalOfArticles: 1 },
        },
      });
    });
    return store.dispatch(getMyArticles()).then(() => {
      expect(store.getActions().length).toEqual(1);
    });
  });
});
