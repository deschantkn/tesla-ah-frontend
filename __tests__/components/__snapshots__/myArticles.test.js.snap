// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetch All Articles tests... it should render the AllArticles Component 1`] = `ShallowWrapper {}`;

exports[`Fetch All Articles tests... it should render the AllArticles Component 2`] = `ShallowWrapper {}`;
